pkg/asset/agent/image/agentimage.go

```diff
@@ -80,3 +80,8 @@ func (a *AgentImage) Files() []*asset.File {
 	// Return empty array because File will never be loaded.
 	return []*asset.File{}
 }
+
+func (a *AgentImage) PersistToFile(directory string) error {
+	// Do some or all of the stuff from Generate here
+	return nil
```

Comment:
:+1: It would be easy enough to store the io stream in the struct and just write it to disk (ll. 51-63) here.